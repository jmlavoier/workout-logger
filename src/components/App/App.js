import React from 'react';

import Form from 'containers/Form';
import WorkoutList from 'containers/WorkoutList';

const App = () => (
  <div>
    <h1>Workout Logger</h1>
    <Form />
    <WorkoutList />
  </div>
);

export default App;