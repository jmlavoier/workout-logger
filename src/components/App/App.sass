@import 'css/variables'
  
.app
  background-color: $color-white-1
  height: 100vh
  margin: 0

  h2
    margin-top: 0
    margin-bottom: 0
    padding-top: 10px
    padding-bottom: 10px
    background: $color-blue-default
    color: $color-white-1
    text-align: center

  @media screen and (min-width: 769px)
    background: linear-gradient(to top right, #a1c4fd 5%, #c2e9fb 65%)
    background-image: linear-gradient(to top right, #a1c4fd 5%, #c2e9fb 65%)
    padding-top: 20px