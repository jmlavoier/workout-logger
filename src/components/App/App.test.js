import React from 'react';
import { shallow, mount } from 'enzyme';

import App from './App';

describe('<App/>', () => {
  it('Should component match snapshot', () => {
    const wrapper = mount(<App/>);
  
    expect(wrapper).toMatchSnapshot();
  });
});
