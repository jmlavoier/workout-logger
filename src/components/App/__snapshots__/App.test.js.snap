// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> Should component match snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <App>
    <div>
      <h1>
        Workout Logger
      </h1>
      <Connect(Form)>
        <Form
          changeField={[Function]}
          clickAdd={[Function]}
          form={
            Object {
              "date": "2018-04-20",
              "timeSpent": "0:45",
              "workoutType": "Run",
            }
          }
        >
          <Container>
            <div>
              <Box>
                <div>
                  <InputTime
                    name="time-spent"
                    onChange={[Function]}
                    value="0:45"
                  >
                    <input
                      onChange={[Function]}
                      type="text"
                      value="0:45"
                    />
                  </InputTime>
                </div>
              </Box>
              <Box>
                <div>
                  <Select
                    id=""
                    name="workout-type"
                    onChange={[Function]}
                    options={
                      Array [
                        Object {
                          "id": "1",
                          "value": "Run",
                        },
                        Object {
                          "id": "2",
                          "value": "Swimming",
                        },
                        Object {
                          "id": "3",
                          "value": "Bike",
                        },
                        Object {
                          "id": "4",
                          "value": "Stretch",
                        },
                        Object {
                          "id": "5",
                          "value": "Yoga",
                        },
                        Object {
                          "id": "6",
                          "value": "Gym",
                        },
                      ]
                    }
                    value="Run"
                  >
                    <select
                      id=""
                      name="workout-type"
                      onChange={[Function]}
                      type="text"
                      value="Run"
                    >
                      <option
                        key="1"
                        value="Run"
                      >
                        Run
                      </option>
                      <option
                        key="2"
                        value="Swimming"
                      >
                        Swimming
                      </option>
                      <option
                        key="3"
                        value="Bike"
                      >
                        Bike
                      </option>
                      <option
                        key="4"
                        value="Stretch"
                      >
                        Stretch
                      </option>
                      <option
                        key="5"
                        value="Yoga"
                      >
                        Yoga
                      </option>
                      <option
                        key="6"
                        value="Gym"
                      >
                        Gym
                      </option>
                    </select>
                  </Select>
                </div>
              </Box>
              <Box>
                <div>
                  <InputTime
                    name="date"
                    onChange={[Function]}
                    value="2018-04-20"
                  >
                    <input
                      onChange={[Function]}
                      type="text"
                      value="2018-04-20"
                    />
                  </InputTime>
                </div>
              </Box>
              <Box>
                <div>
                  <Button
                    onClick={[Function]}
                  >
                    <button
                      onClick={[Function]}
                    >
                      Add
                    </button>
                  </Button>
                </div>
              </Box>
            </div>
          </Container>
        </Form>
      </Connect(Form)>
      <Connect(WorkoutList)>
        <WorkoutList
          changeOrder={[Function]}
          workoutList={Array []}
        >
          <Container>
            <div>
              <Box>
                <div>
                  <List
                    changeOrder={[Function]}
                    items={Array []}
                  >
                    <table>
                      <thead>
                        <tr>
                          <td>
                            Tempo
                          </td>
                          <td>
                            Tipo
                          </td>
                          <td>
                            Data
                          </td>
                        </tr>
                      </thead>
                      <tbody />
                    </table>
                  </List>
                </div>
              </Box>
            </div>
          </Container>
        </WorkoutList>
      </Connect(WorkoutList)>
    </div>
  </App>
</Provider>
`;
